src/idlcxx/src/streamers.c

```diff
@@ -543,10 +654,26 @@ generate_streamers(const idl_pstate_t* pstate, struct generator *gen)
   idl_visitor_t visitor;
   const char *sources[] = { NULL, NULL };
 
-  memset(&streams, 0, sizeof(streams));
+  setup_streams(&streams, gen);
+
+  memset(&visitor, 0, sizeof(visitor));
+
+  visitor.visit = IDL_TYPEDEF | IDL_MODULE;
+  visitor.accept[IDL_ACCEPT_TYPEDEF] = &process_typedef;
+  visitor.accept[IDL_ACCEPT_MODULE] = &process_module;
+  assert(pstate->sources);
+  sources[0] = pstate->sources->path->name;
+  visitor.sources = sources;
```

Comment:
Visiting twice seems odd?

Reply:
This was done so that the streaming functions for the typedefs would be in their own namespace, but those for the structs and unions would not be.
I chose this because the typedefs need this to be able to distinguish between different typedefs which map onto the same C++ class, which could cause the compiler to not be able to determine which of the streaming functions to use because they have the same signature.
For the structs and unions, the streaming function signatures are because the parameter which is part of the function signature is unique.

The main reason was to not have to make very long namespaces for calling the streaming functions of non-typedef instances.

Reply:
I think I'm fine with it for this PR, but I'd prefer the functions for constructed types be placed in their own namespace as well. Same goes for the functions we use to serialize strings, sequences, etc. They're currently in the global namespace, which is bad practice when using C++ if you ask me. We'll tackle it in a follow-up PR.

Reply:
I have added all the points that still need improvement and work to issue #83